src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -0,0 +1,170 @@
+
+        self.userCert = cert()
+        self.userKey = ckey()
+        self.rucio = Rucio(acct=self.rucioAcct,
+                           hostUrl=self.rucioUrl,
+                           authUrl=self.rucioAuthUrl)
+                           # configDict={'logger': self.logger})
+
+    @timeFunction
+    def algorithm(self, parameters=None):
+        """
+        Executed in every polling cycle. The actual logic of the component.
+        :param parameters: not really sure what those can be(!)
+        :return: only what is returned by the decorator
+        """
+        logging.info("Running Workflow updater injector poller algorithm...")
+        try:
+            # retrieve list of workflows with unfinished Production
+            # or Processing subscriptions
+            wflowSpecs = self.listActiveWflows.execute()
+            if not wflowSpecs:
+                logging.info("Agent has no active workflows at the moment")
+                return
+
+            # figure out workflows that have pileup
+            puWflows = self.findWflowsWithPileup(wflowSpecs)
+
+            # otherwise, move on retrieving pileups
+            pileupMap = self.getPileupDocs()
+        except Exception as ex:
+            msg = "Caught unexpected exception in WorkflowUpdater. Details:\n%s" % str(ex)
+            logging.exception(msg)
+            raise WorkflowUpdaterException(msg) from None
+
+    def getPileupDocs(self):
+        """
+        Fetch all pileup documents from MSPileup and preprocess the data.
+        :return: a list of dictionaries in the following format:
+          {"pileupName": string with pileup name,
+           "customName": string with custom pileup name - if any,
+           "rses": list of RSE names,
+           "blocks": list of block names}
+        """
+        mgr = RequestHandler()
+        headers = {'Content-Type': 'application/json'}
+        data = mgr.getdata(self.msPileupUrl, params={}, headers=headers, verb='POST',
+                           ckey=self.userKey, cert=self.userCert, encode=True, decode=True)
+        if data and data.get("result", []):
+            if "error" in data["result"][0]:
+                msg = f"Failed to retrieve MSPileup documents. Error: {data}"
+                raise WorkflowUpdaterException(msg)
+
+        logging.info("A total of %d pileup documents have been retrieved.", len(data["result"]))
+        pileupMap = []
+        for puItem in data["result"]:
+            logging.info("Pileup: %s, custom name: %s, expected at: %s, but currently available at: %s",
+                         puItem['pileupName'], puItem['customName'],
+                         puItem['expectedRSEs'], puItem['currentRSEs'])
+            thisPU = {"pileupName": puItem['pileupName'],
+                      "customName": puItem['customName'],
+                      "rses": puItem['currentRSEs'],
+                      "blocks": []}
```

Comment:
I see that blocks all the time will be an empty list, why do you need to construct it here?

Reply:
My goal is to populate this variable with a list of block names in a future pull request. Just so we have a data structure with all the relevant information for a given pileup dataset.

Reply:
In this case please change docstring accordingly to avoid confusion. For instance, I suggest to make concrete estimates using approximate time spent in rucio, e.g. `O(1-10sec)`, multiplied by total number of given workflows, i.e. how much time we'll spend for 100 workflows (which is what I expect as average).

Reply:
I updated the docstring. I can also add a comment of expected time to be spent on that, but as things change, I am sure it will simply become outdated and not meaningful.